src/js/files/components/UploadItem.js

```diff
@@ -39,3 +41,42 @@ export const UploadItem = ({ name, progress, size }) => (
         </UploadItemTitle>
     </StyledUploadItem>
 );
+
+const StyledUploadItemError = styled(StyledUploadItem)`
+    i.fa-times {
+        color: ${props => getColor({ theme: props.theme, color: "red" })};
+        font-size: 20px;
+    }
+    span: last-child {
+        color: ${props => getColor({ theme: props.theme, color: "red" })};
+        font-size: 12px;
+        font-style: italic;
+    }
+    i.fa-trash {
+        margin-left: 5px;
+        font-size: 14px;
+    }
+`;
+
+export const UploadItemError = ({ name, onRemove, localId }) => (
+    <StyledUploadItemError>
+        <AffixedProgressBar now={100} color="red" />
+        <UploadItemTitle>
+            {<Icon name="times" />}
+            {name}
+            <span>
+                Failed <Icon aria-label={`delete ${name}`} name="trash" color="red" onClick={() => onRemove(localId)} />
+            </span>
+        </UploadItemTitle>
+    </StyledUploadItemError>
```

Comment:
Combine this with `UploadItem`. What was the strategy around having separate components for this?

Reply:
When implementing I figured that two simple components would be easier to understand and possibly faster than one monolithic component. I have uploaded a new version with just one UploadItem, so you can make the call on which you think is better.

Reply:
I appreciate the abstraction offered by a single `UploadItem` component. You just pass it props from the upload object and it knows what to do.